jetty-server/src/test/java/org/eclipse/jetty/server/RequestTest.java

```diff
@@ -872,6 +873,29 @@ public boolean check(HttpServletRequest request, HttpServletResponse response)
         assertEquals(" x=z; ", results.get(i++));
     }
 
+    @Test
+    public void testConnectRequestURL() throws Exception
+    {
+        final AtomicReference<String> resultRequestURL = new AtomicReference<>();
+        final AtomicReference<String> resultRequestURI = new AtomicReference<>();
+        _handler._checker = (request, response) ->
+        {
+            resultRequestURL.set("" + request.getRequestURL());
+            resultRequestURI.set("" + request.getRequestURI());
+            return true;
+        };
+
+        String rawResponse = _connector.getResponse(
+            "CONNECT myhost:9999 HTTP/1.1\n" +
+                "Host: myhost:9999\n" +
```

Comment:
What happens when `Host` is different from the authority-form just after the `CONNECT` method?
Can we write another test case to document how we should behave in that case?

Reply:
Right now in Jetty 9.4.48, if the request is say ....

```
CONNECT myhost:9999 HTTP/1.1
Host: otherhost:8888
Connection: close
```

The two API calls become ...

* `HttpServletRequest.getRequestURL()` becomes `http://otherhost:8888`
* `HttpServletRequest.getRequestURI()` becomes `otherhost:8888`

The information `myhost:9999` actually gets lost.
When the request gets parse, the request HttpURI starts out with `myhost:9999` (authority) when the `CONNECT` line is parsed.
Then gets overridden with `otherhost:8888` when the `Host` header is parsed.

Prior to PR #8014, the details in `myhost:9999` was actually stored in `HttpURI.getPath()` so that the two different authorities could be tracked separately.

Reply:
RFC 7230, section 5.4 states:

> If the target URI includes an authority component, then a
   client MUST send a field-value for Host that is identical to that
   authority component

That does not explicitly say "the request target is an authority-form", but perhaps could be inferred.

Also (same section):

> When a proxy receives a request with an absolute-form of
   request-target, the proxy MUST ignore the received Host header field
   (if any) and instead replace it with the host information of the
   request-target.

So the `Host` header value can be ignored, as long as there is another authority.
Again, it does not explicitly say the behavior for the authority-form, but talks about proxies and I guess we could have the same behavior (i.e. in case of authority-form, we ignore `Host` and use the authority-form).

If we change `getServerName()` to take that behavior into account (i.e. in case of absolute-form and authority-form we don't use the `Host` header, but the given authority instead), do we break much?